hack/check-plutono-dashboards.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+#
+# Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+echo "> Checking Plutono dashboards"
+
+
+function check_dashboards {
+  while IFS= read -r file
+  do 
+    jq -c -r '{title: (.title // error("title is not set")),
+               uid:   (.uid   // error("uid is not set"))}
+              | "\(input_filename) \(.uid)"' "$file" \
+    || { echo "Error: Failure while parsing dashboard: $file" >&2; return 1; }
+  done < <(find . -path '*/dashboards/*' -name '*.json' -type f) \
+  | sort -k 2 | uniq -D -f 1 \
+  | { grep . && echo "Error: dashboards with duplicate UIDs" >&2 && return 1 || return 0; }
+}
```

Comment:
```suggestion
function check_dashboards {
  find . -path '*/dashboards/*' -name '*.json' -type f \
  | while IFS= read -r file; do
      jq -c -r '{title: (.title // error("title is not set")),
                 uid:   (.uid   // error("uid is not set"))}
                | if (.uid | length) > 40
                    then error("uid is too long (max length is 40 characters): \(.uid)")
                    else .
                  end
                | "\(input_filename) \(.uid)"' "$file" \
      || { echo "Error: Failure while parsing dashboard: $file" >&2; return 1; }
    done \
  | sort -k 2 | uniq -D -f 1 \
  | { grep . && echo "Error: dashboards with duplicate UIDs" >&2 && return 1 || return 0; }
}
```

I improved the snippet that I initially provided:
- the find block comes first, maybe it is easier to read the script this way: first find the files in the dashboards folders, then parse them with jq, ... Previously the find block came after the done keyword and hence the data flow was not so natural.
- I added one more assertion: the uid should not be longer than 40 characters

Reply:
In general, @Sallyan, please make sure to test your PR in a local setup so that such issues can be found by you.

Reply:
@istvanballok Thanks for the update. 
```
'{title: (.title // error("title is not set")), uid: (.uid // error("uid is not set"))}'
``` 
This part jq filter expression validates either title or uid field is missing. I add new code lines to check title or uid field also cannot be empty. 
```
      if jq -e '.title == "" or .uid == ""' "$file" >/dev/null; then
          echo "Error: Title or UID is empty in dashboard: $file" >&2
          return 1
      fi
```